src/firefly/js/ui/CheckboxGroupInputField.jsx

```diff
@@ -62,33 +62,67 @@ CheckboxGroupInputFieldView.propTypes= {
     }),
 };
 
-
+export const SelectAllCheckbox = memo(({ fieldKey, label = 'Select all', groupKey }) => {
+    return (
+        <Stack direction='row' alignItems='center'>
+            <CheckboxGroupInputField
+                key={`${fieldKey}_all`}
+                fieldKey={fieldKey}
+                groupKey={groupKey}
+                initialState={{
+                    value: '', //empty initially, _all_ will be toggled as needed
+                    label: ''
+                }}
+                options={[{label, value: '_all_' }]}
+            />
+        </Stack>
+    );
+});
 
 function handleOnChange(ev, viewProps, fireValueChange) {
-    // when a checkbox is checked or unchecked
-    // the array, representing the value of the group,
-    // needs to be updated
-    const value= convertValue(viewProps.value,viewProps.options);
-
     const val = ev.target.value;
     const checked = ev.target.checked;
-    const curValueArr = getCurrentValueArr(value);
-    const idx = curValueArr.indexOf(val);
-    if (checked) {
-        if (idx < 0) curValueArr.push(val); // add val to the array
+    const allValues = (viewProps.options ?? []).map((option) => option.value);
+
+    //expand '_all_' into the full options list and use that as the list of curr values array
+    const normalizedValue = convertValue(viewProps.value, viewProps.options);
+    const curValueArr = getCurrentValueArr(normalizedValue);
```

Comment:
Logic should not be here.  It should be in the `SelectAllCheckbox` component.

Reply:
Ok yes I understand this now. Separation of concern as well! I'll implement this.